gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/jupiter/handlers/api/SyncApiReactor.java

```diff
@@ -253,7 +279,7 @@ private Completable invokeBackend(final RequestExecutionContext ctx) {
                 }
             )
             .doOnSubscribe(disposable -> ctx.request().metrics().setApiResponseTimeMs(System.currentTimeMillis()))
-            .doOnTerminate(
+            .doFinally(
```

Comment:
I don't think we should change this cause the api response time must be calculated before processing the rest of the chain.

The doFinally action is executed after the signal has been propagated 
![image](https://user-images.githubusercontent.com/5955401/180803209-da0ba83b-3f9b-40ab-88c7-0221d1d59a2e.png)

whereas the doOnTerminate ensure that it will be executed before

![image](https://user-images.githubusercontent.com/5955401/180803070-1f127193-b485-481e-9d63-390d51825738.png)

Reply:
indeed, timeout use a dispose 
Solution is to combine onTerminate and onDispose

Reply:
resolved by [23d903c](https://github.com/gravitee-io/gravitee-api-management/pull/2224/commits/23d903cbace24f58a27b5caac671cfd359bf9d83)